python/python/tests/test_fork.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+
+from pathlib import Path
+import lance
+import os
+import pyarrow as pa
+import pyarrow.compute as pc
+
+N_DIMS = 768
+NUM_ROWS = 100_000
+NEW_ROWS = 10_000
+
+def create_table(num_rows) -> pa.Table:
+    return pa.table(
+        {
+            "a": pc.random(num_rows).cast(pa.float32()),
+            "b": pa.array(range(0, num_rows)),
+        }
+    )
+
+def test_table_roundtrip(tmp_path: Path):
+    uri = tmp_path
+
+    tbl = create_table(100)
+    lance.write_dataset(tbl, uri)
+
+    os.fork()
```

Comment:
How does fork work with pytest?

Reply:
pytest doesn't seem to mind forking. Since multiprocessing is supported in pytest I expect there are others using fork here.